plugins/filter_aws/aws.c

```diff
@@ -31,22 +31,39 @@
 #include <fluent-bit/flb_io.h>
 #include <fluent-bit/flb_kv.h>
 
+
 #include <monkey/mk_core/mk_list.h>
 #include <msgpack.h>
-
 #include <stdlib.h>
 #include <errno.h>
 
 #include "aws.h"
 
 static int get_ec2_token(struct flb_filter_aws *ctx);
-static int get_metadata(struct flb_filter_aws *ctx, char *metadata_path,
-                        flb_sds_t *metadata, size_t *metadata_len);
+static int get_metadata(struct flb_filter_aws *ctx, char *metadata_path, flb_sds_t *metadata, size_t *metadata_len);
+static int get_metadata_by_key(struct flb_filter_aws *ctx, char *metadata_path,
+                        flb_sds_t *metadata, size_t *metadata_len, char *key);
+
+static int set_enabled_fields(const char *key, struct flb_filter_instance *f_ins, int *new_keys_number)
+{
+    const char *tmp = NULL;
+    int key_enabled = FLB_FALSE;
+    tmp = flb_filter_get_property(key, f_ins);
```

Comment:
You can change/get rid of this function by using `offsetof` in the config map as shown in one of my other comments. It will set the config property to true/false for you when you set the config map, for example as shown here: https://github.com/fluent/fluent-bit/blob/master/plugins/out_s3/s3.c#L190

If you do that, the only remaining task is to count the number of enabled fields.

Reply:
Yeah good call! Removed the function.